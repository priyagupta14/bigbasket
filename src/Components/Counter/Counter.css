.product-counter {
  display: flex;
  justify-content: space-between;
  background-color: rgb(197, 190, 190);
  height: 3rem;
}
.product-counter > p {
  height: 1rem;
  width: 5rem;
}
.product-counter > button {
  width: 2rem;
}
