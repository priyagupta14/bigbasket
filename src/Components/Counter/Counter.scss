@import '../../styles/index.scss';

.product-counter {
  display: flex;
  justify-content: space-between;
  border: 1px solid black;
  background-color: $primary-background-color;
  height: 3rem;
}
.zeroProduct{
  background-color: rgb(148, 147, 147);
  display: flex;
  justify-content: space-between;
  border: 1px solid black;
  background-color: $primary-background-color;
  height: 3rem;
}
.moreProduct{
  display: flex;
  justify-content: space-between;
  border: 1px solid black;
  background-color: $primary-background-color;
  height: 3rem;
  background-color: rgb(211, 198, 123);
}
.product-counter > p {
  height: 1rem;
  width: 5rem;
  
}
.product-counter > button {
  width: 2rem;
  height: auto;
}
