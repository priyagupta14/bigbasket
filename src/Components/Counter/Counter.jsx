import React, { Component } from "react";
import "./Counter.css";

export default class Counter extends Component {
  constructor(props) {
    super(props);
  }
  render() {
    return (
      <div class="product-counter">
        <button onClick={this.props.onIncrement}>+</button>
        <p>{this.props.itemCount} in basket</p>
        <button onClick={this.props.onDecrement}>-</button>
      </div>
    );
  }
}
