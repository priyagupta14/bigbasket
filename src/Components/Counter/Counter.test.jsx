/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import {
  fireEvent, render, screen,
} from '@testing-library/react';
import Counter from './Counter';

describe(Counter.name, () => {
  const mockProp = {
    itemCount: 0,
    onDecrement: jest.fn(),
    onIncrement: jest.fn(),
  };
  afterEach(() => {
    jest.clearAllMocks();
  });
  it('should match snapshot', () => {
    const { container } = render(<Counter {...mockProp} />);
    expect(container).toMatchSnapshot();
  });
  it('should display button and content', () => {
    render(<Counter {...mockProp} />);
    const incrementElement = screen.getByText('+');
    expect(incrementElement.tagName).toBe('BUTTON');
    const decrementElement = screen.getByText('-');
    expect(decrementElement.tagName).toBe('BUTTON');
    screen.getByText('0 in basket');
  });
  it('should increment product count', () => {
    render(<Counter {...mockProp} />);
    const incrementElement = screen.getByText('+');
    fireEvent.click(incrementElement);
    expect(mockProp.onIncrement).toHaveBeenCalled();
  });
  it('should decrement product count', () => {
    render(<Counter {...mockProp} />);
    const decrementElement = screen.getByText('-');
    fireEvent.click(decrementElement);
    expect(mockProp.onDecrement).toHaveBeenCalled();
  });
});
