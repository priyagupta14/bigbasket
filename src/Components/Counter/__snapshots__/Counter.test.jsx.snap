// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter should match snapshot 1`] = `
<div>
  <div
    class="zeroProduct"
  >
    <button
      type="button"
    >
              +      
    </button>
    <p>
      0
       
      in basket
    </p>
    <button
      type="button"
    >
      -
    </button>
  </div>
</div>
`;
